domain/model/state/state.go

```diff
@@ -96,6 +102,46 @@ func createModel(ctx context.Context, uuid model.UUID, tx *sql.Tx) error {
 	return nil
 }
 
+// createModelAgent is responsible for create a new model's agent record for the
+// given model UUID. If a model agent record already exists for the given model
+// uuid then an error satisfying [modelerrors.AlreadyExists] is returned. If no
+// model exists for the provided UUID then a [modelerrors.NotFound] is returned.
+func createModelAgent(
+	ctx context.Context,
+	modelUUID model.UUID,
+	agentVersion version.Number,
+	tx *sql.Tx,
+) error {
+	stmt := `
+INSERT INTO model_agent (model_uuid, previous_version, target_version)
+    VALUES (?, ?, ?)
+`
+
+	res, err := tx.ExecContext(ctx, stmt, modelUUID, agentVersion.String(), agentVersion.String())
+	if jujudb.IsErrConstraintPrimaryKey(err) {
+		return fmt.Errorf(
+			"%w for uuid %q while setting model agent version",
+			modelerrors.AlreadyExists, modelUUID,
+		)
+	} else if jujudb.IsErrConstraintForeignKey(err) {
```

Comment:
This only works because there's a single FK. What happens if we introduce another one later. We don't know which one is violated do we? Not a blocker but just sharing a thought.

Reply:
Yep, it's a problem I have had to deal with in other places. You need to either pull apart the error or break the operations up. Just have to find a way to deal with it. The error message has a great deal of value in my opinion so it's worth handling correctly.